### PR TITLE
[qfix] Save generated kernel interface name

### PR DESCRIPTION
**Pull request by Bolodya1997**

### CODE REVIEW

**Review by denis-tingaikin — APPROVED** [review]

File: pkg/api/networkservice/mechanisms/kernel/helpers.go [hunk 1]
```diff
@@ -100,7 +100,7 @@ func (m *Mechanism) GetInterfaceName(conn *networkservice.Connection) string {
 		if len(name) > LinuxIfMaxLength {
 			name = name[:LinuxIfMaxLength]
 		}
-		return name
+		m.GetParameters()[InterfaceNameKey] = name
```
Comment (edwarnicke) [inline pkg/api/networkservice/mechanisms/kernel/helpers.go:103]: Why is this needed?

Comment (denis-tingaikin) [reply pkg/api/networkservice/mechanisms/kernel/helpers.go:103]: Vlad found that in case of healing, we could re-generate the name for the interface and add it again.

@Bolodya1997 Could you explain it in detail?

Comment (Bolodya1997) [reply pkg/api/networkservice/mechanisms/kernel/helpers.go:103]: probably it is not fully correct to name the case I have tested as **NSMgr restart**, it is most actually a **NSMgr is getting replaced with another NSMgr with the same unix/IP addresses** - so the difference between is that another NSMgr has another name
in production we should get this case when we use rolling update on the NSMgr daemonset template:
* update NSMgr version
* change NSMgr envs
so it is not actually a restart, but it also should be healed as a restore

in such case new NSMgr overwrites old NSMgr path segments, so we have:
```
NSC -> NSMgr(1) -> Forwarder -> NSMgr(2) -> NSE
```
turning to
```
NSC -> new NSMgr(1) -> Forwarder -> new NSMgr(2) -> NSE
```
so when Forwarder starts trying to use `new NSMgr(2)` instead of `NSMgr(2)` for generating the link name, it generates a new name and receives a `Link not found` error and so fails a restore

Comment (denis-tingaikin) [reply pkg/api/networkservice/mechanisms/kernel/helpers.go:103]: @edwarnicke Could you share your thoughts on why previously we do not store generated name in the parameters?
